Add an option to disable the generation of debug symbols

Whissi opened the pull request:
Not every user is interested in building software with debug symbols.
If the user has set the new "--without-debug-symbols" option, mklove won't
set the appropriate compiler flag to generate debug symbols anymore.

Fixes https://github.com/edenhill/mklove/issues/7

Whissi commented:
Not sure about the help text. My intention was to express that it is the mklove framework which sets the `-g` flag automatically, not an option from the project which is using the mklove framework...

The commit itself contains the whole block because I also fixed some coding style issues (mixed tabs with spaces,,, now the block only uses spaces. I think that's the default).

edenhill commented:
Looks good, but can you rename it to '--enable/disable-debug-symbols' ?  I think that is more in line with the other toggle options.

Whissi commented:
I updated the PR. I renamed the option to `--disable-debug-symbols` and updated the help text to the text [Mozilla](http://mxr.mozilla.org/mozilla/source/nsprpub/configure.in#359) is using.

edenhill commented:
Thank you!

edenhill commented:
Will you submit another PR for updating librdkafka's mklove according?
Extra points for also doing kafkacat

Whissi commented:
Yep, I'll prepare a PR for librdkafka. But I have more patches in my pipeline I want to check first (need more testing before I'll post them). Expect the PR within the next 4 days.

Regarding kafkacat: I haven't to look at code yet but shouldn't be a problem.
